code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm

```diff
@@ -4,26 +4,50 @@
 	name = "Nutriment"
 	id = "nutriment"
 	description = "All the vitamins, minerals, and carbohydrates the body needs in pure form."
+	taste_mult = 4
 	reagent_state = SOLID
 	metabolism = REM * 4
 	var/nutriment_factor = 30 // Per unit
 	var/injectable = 0
 	color = "#664330"
 
+/datum/reagent/nutriment/mix_data(var/list/newdata, var/newamount)
+
+	if(!islist(newdata) || !newdata.len)
+		return
+	for(var/i in 1 to newdata.len)
+		if(!(newdata[i] in data))
+			data.Add(newdata[i])
+			data[newdata[i]] = 0
+		data[newdata[i]] += newdata[newdata[i]]
+	var/totalFlavor = 0
+	for(var/i in 1 to data.len)
+		totalFlavor += data[data[i]]
+	for(var/i in 1 to data.len) //cull the tasteless
+		if(data[data[i]]/totalFlavor * 100 < 10)
+			data[data[i]] = null
+			data -= data[i]
+			data -= null
+
 /datum/reagent/nutriment/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
 	if(!injectable)
-		M.adjustToxLoss(0.1 * removed)
+		M.adjustToxLoss(0.2 * removed)
 		return
 	affect_ingest(M, alien, removed)
 
 /datum/reagent/nutriment/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
-	switch(alien)
-		if(IS_UNATHI) removed *= 0.5
```

Comment:
This was very specifically added to make Unathi gain less nutrition from non-meat sources.